vm-virtio/src/lib.rs

```diff
@@ -132,14 +203,40 @@ impl Translatable for u64 {
 }
 
 /// Helper for cloning a Queue since QueueState doesn't derive Clone
-pub fn clone_queue(queue: &Queue) -> Queue {
+pub fn clone_queue(
+    queue: &Queue,
+    #[cfg(all(feature = "mshv", feature = "sev_snp"))] access_platform: Option<
+        Arc<dyn AccessPlatform>,
+    >,
+) -> Queue {
     let mut q = Queue::new(queue.max_size()).unwrap();
 
     q.set_next_avail(queue.next_avail());
     q.set_next_used(queue.next_used());
     q.set_event_idx(queue.event_idx_enabled());
     q.set_size(queue.size());
     q.set_ready(queue.ready());
+
+    #[cfg(all(feature = "mshv", feature = "sev_snp"))]
+    {
+        let desc_addr = GuestAddress(queue.desc_table());
+        let avail_addr = GuestAddress(queue.avail_ring());
+        let ring_addr = GuestAddress(queue.used_ring());
+
+        desc_addr.translate_gva(
```

Comment:
Since this is just cloning an existing queue - why aren't haven't those addresses already been through this?